### ISSUE TITLE
[Feature Request]: Allow writing/appending to `installationScript`

### ISSUE THREAD

**Issue by kai-tub** [cachix/git-hooks.nix#476]
Hey :wave:,

Thank you very much for working on this awesome project!
I gave up on using `pre-commit` some time ago due to weird environment issues across different machines.
With the power of nix, it works like a charm!

But I currently face an issue/limitation that is "inherited" by from `pre-commit` (https://github.com/pre-commit/pre-commit/issues/693).

I am using `git-lfs` and to push to the remote LFS server, one usually installs the `pre-push` hook via `git lfs install`, which essentially writes:

```bash
#!/bin/sh
command -v git-lfs >/dev/null 2>&1 || { echo >&2 "\nThis repository is configured for Git LFS but 'git-lfs' was not found on your path. If you no longer wish to use Git LFS, remove this hook by deleting the 'pre-push' file in the hooks directory (set by 'core.hookspath'; usually '.git/hooks').\n"; exit 2; }
git lfs pre-push "$@"
```

The issue with `pre-commit` is that it doesn't allow a `raw` execution mode that forwards the arguments from the `git` hook call. 
As discussed in https://github.com/pre-commit/pre-commit/issues/693, some workaround this issue by "wrapping" their hook to execute the above script as well.

One can test this by updating the `.git/hooks/pre-push` file after `pre-commit install` was run to include the `git lfs pre-push "@"` line (optionally with nix-path).

From what I can see, it would be nice if the following block:

https://github.com/cachix/git-hooks.nix/blob/f451c19376071a90d8c58ab1a953c6e9840527fd/modules/pre-commit.nix#L327-L395

could be writeable or have `extraPre`/`extraPost` data to provide an escape hatch to allow "injecting" additional code.

I fully understand that this is a limitation from pre-commit but since this repository also handles the "installation" of the pre-commit hooks, I believe that supporting/allowing such corner-cases would be beneficial.

Maybe there is a simple way to correctly configure git-lfs to play nicely with pre-commit and I am just not seeing it :thinking: 

Either way, thanks for considering it!

**Comment by kai-tub** [comment]
Nah.

Sorry. :bow: 

I have thought about this more and missed that this is essentially executed during `enterShell` or `shellHook` and that one can append the "update" Script to the activation environment.
My main misunderstanding was that the hook files (`.git/hooks/pre-push` for example) are _not_ tracked inside of the nix-store but are generated during the environment activation (or when the `nix flake check` command is run). So, there is no issue in "patching/updating" these files in a separate step.

I also believe there is no good place to document this peculiar possible workaround for `git-lfs`.

TL;DR Sorry for the noise!

**Comment by shackra** [comment]
@matdibu I achieve this with a custom python script and some nix magic, you can check this commit in my project here https://github.com/shackra/jorgeayara.dev/commit/208d936b617738a1c56e6d984f47bbf72af62a11
